sound/soc/sof/intel/Makefile

```diff
@@ -6,7 +6,7 @@ snd-sof-acpi-intel-bdw-objs := bdw.o
 snd-sof-intel-hda-common-objs := hda.o hda-loader.o hda-stream.o hda-trace.o \
 				 hda-dsp.o hda-ipc.o hda-ctrl.o hda-pcm.o \
 				 hda-dai.o hda-bus.o \
-				 apl.o cnl.o tgl.o icl.o
+				 apl.o cnl.o tgl.o icl.o hda-common-ops.o
```

Comment:
this looks good @plbossart but what is missing is the invocation of ops_init() from probe() right?

Reply:
yes, that's a remaining open, I wasn't sure if this belongs in the core or in the platform probe.

Reply:
I think it should be platform probe. The core should only care of the IPC-independent mandatory ops and the individual platform should handle the IPC dependent ops that are optional

Reply:
the platform probe is already part of the common structure, so that won't work?

Reply:
why not? it is IPC independent though right? please give me an hour, let me write up what I have in mind

Reply:
it doesn't matter if it's independent or not, if we want to reuse the same platform_probe across all HDaudio platforms, we need to set the pointer from core.c, before we check for all mandatory callbacks.

Again, my changes are not strictly related to the IPC type, they help save between all 4 versions of Intel platforms. see apl.c and friends, they are way smaller now with no duplication across variants.